crates/holochain/src/core/sys_validate/error.rs

```diff
@@ -133,8 +133,6 @@ pub enum ValidationOutcome {
     NotCreateLink(ActionHash),
     #[error("The action was expected to be a new entry action but was {0:?}")]
     NotNewEntry(Action),
-    #[error("The dependency {0:?} is not held")]
-    NotHoldingDep(AnyDhtHash),
```

Comment:
This was miss-mapped and wasn't used consistently. In most places it was actually a `DepMissingFromDht` you'd get. I think it's simpler to just have the one even though these are technically two distinct scenarios. The meaning is the same in this context, we don't know whether the op exists until we can get it from somewhere.

Reply:
I really appreciate this and all comments you make on your PRs, it really helps to understand the rationale.

And yes without diving into the code myself I find it hard to imagine we'd treat either of those cases differently, so that sounds good.